# 4.6.1 Combining & Exporting Data

### This script contains the following points:

#### 01. Importing Libraries
#### 02. Importing Data & Checking Dimensions
#### 03. Merging Dataframes
#### 04. Using the Indicator Argument to Verify Full Match
#### 05. Export Combined Data in Pickle Format
#### 06. Export Modified df_ords as CSV

## 01. Importing Libraries

In [1]:
# Import libraries
import pandas as pd
import numpy as np
import os

## 02. Importing Data & Checking Dimensions

In [2]:
# Turn Path into a String & Assign Python Shortcut
path = '/Users/mgottlieb/Documents/Instacart Basket Analysis'

In [7]:
# Import the orders_products_prior.csv dataset
df_ords_prior = pd.read_csv(os.path.join(path, '02 Data', 'Original Data', 'orders_products_prior.csv'), index_col = False)

In [8]:
# Check the output
df_ords_prior.head()

order_id  product_id  add_to_cart_order  reordered
0         2       33120                  1          1
1         2       28985                  2          1
2         2        9327                  3          0
3         2       45918                  4          1
4         2       30035                  5          0

In [13]:
# Import the orders_checked.csv dataset
df_ords = pd.read_csv(os.path.join(path, '02 Data', 'Prepared Data', 'orders_checked.csv'), index_col = False)

In [14]:
# Check the output
df_ords.head()

Unnamed: 0.1  Unnamed: 0  order_id  user_id evaluation_status  \
0             0           0   2539329        1             prior   
1             1           1   2398795        1             prior   
2             2           2    473747        1             prior   
3             3           3   2254736        1             prior   
4             4           4    431534        1             prior   

   order_number  orders_day_of_week  order_hour_of_day  \
0             1                   2                  8   
1             2                   3                  7   
2             3                   3                 12   
3             4                   4                  7   
4             5                   4                 15   

   days_since_prior_order  days_since_prior_order_missing  
0                     7.0                               1  
1                    15.0                               0  
2                    21.0                               0  
3                    29.0                               0  
4                    28.0                               0

In [11]:
df_ords_prior.shape

(32434489, 4)

In [15]:
df_ords.shape

(3421083, 10)

## 03. Merging Dataframes

In [16]:
df_merged_large = df_ords.merge(df_ords_prior, on = 'order_id', indicator = True)

In [17]:
df_merged_large.head()

Unnamed: 0.1  Unnamed: 0  order_id  user_id evaluation_status  \
0             0           0   2539329        1             prior   
1             0           0   2539329        1             prior   
2             0           0   2539329        1             prior   
3             0           0   2539329        1             prior   
4             0           0   2539329        1             prior   

   order_number  orders_day_of_week  order_hour_of_day  \
0             1                   2                  8   
1             1                   2                  8   
2             1                   2                  8   
3             1                   2                  8   
4             1                   2                  8   

   days_since_prior_order  days_since_prior_order_missing  product_id  \
0                     7.0                               1         196   
1                     7.0                               1       14084   
2                     7.0                               1       12427   
3                     7.0                               1       26088   
4                     7.0                               1       26405   

   add_to_cart_order  reordered _merge  
0                  1          0   both  
1                  2          0   both  
2                  3          0   both  
3                  4          0   both  
4                  5          0   both

## 04. Using the Indicator Argument to Verify Full Match

In [18]:
df_merged_large['_merge'].value_counts()

_merge
both          32434489
left_only            0
right_only           0
Name: count, dtype: int64

The value_counts output indicates that all 32,434,489 rows from the merged DataFrame were matched entries present in both original data sets (both), with no unmatched entries from either the left (left_only) or right (right_only) data sets.

## 05. Export Combined Data in Pickle Format

In [19]:
# Export data to pkl
df_merged_large.to_pickle(os.path.join(path, '02 Data','Prepared Data', 'orders_products_combined.pkl'))

## 06. Export Modified df_ords as CSV

In [20]:
# Export data to csv
df_merged_large.to_csv(os.path.join(path, '02 Data','Prepared Data', 'orders_products_combined.csv'))